File: Arcade/Intro/18_palindromeRearranging.js
function palindromeRearranging(inputString) {
    return (singles(inputString) === inputString.length % 2);
}

function singles(s) {
    var string = s.split('').sort();
    var count = 0;
    while (string.length > 1) {
        if (string[0] === string[1]) {
            string.shift();
            string.shift();
        } else {
            count++;
            string.shift();
        }
    } return string.length === 1 ? count + 1 : count;
}

File: Arcade/Intro/C++/18_palindromeRearranging.cpp
bool palindromeRearranging(std::string inputString) {
    map<char, int> count;
    bool isOdd=false;
    for(char c:inputString){
        count[c]++;
    }
    for(auto i=count.begin();i!=count.end();i++){
        if(i->second%2!=0) {
            if(!isOdd){
                isOdd=true;
            } else return false;
        }
    }
    return true;
}
